From 52459556152533353fe2c53d68cddfe368b5fb15 Mon Sep 17 00:00:00 2001
From: terminills <tdeters@gmail.com>
Date: Fri, 24 Jul 2026 15:19:58 -0500
Subject: [PATCH 306/638] cef: expose WebView resize background color

Apply CEF's upstream set_resize_background_color patch as a discrete prerequisite for CefBrowserView background updates.

AI_PHASE: chromium-cef-bootstrap

AI_STATUS: upstream-webview-background-prerequisite-applied

AI_ASSUMPTION: CEF 52dac911 patch stack is authoritative for Chromium 124.0.6367.0 integration

AI_VALIDATION: libcef_static reported missing views::WebView::SetResizeBackgroundColor

AI_EVIDENCE: test-artifacts/-cef-bootstrap-20260724/13-libcef-static-after-core-cef-patches.log

AI_DECISION_CHAIN:; 52dac911; 6dcd24c759
---
 ui/views/controls/webview/webview.cc | 4 ++++
 ui/views/controls/webview/webview.h  | 4 ++++
 2 files changed, 8 insertions(+)

diff --git a/ui/views/controls/webview/webview.cc b/ui/views/controls/webview/webview.cc
index 112c061004..ee0ea41872 100644
--- a/ui/views/controls/webview/webview.cc
+++ b/ui/views/controls/webview/webview.cc
@@ -159,6 +159,10 @@ void WebView::EnableSizingFromWebContents(const gfx::Size& min_size,
   }
 }
 
+void WebView::SetResizeBackgroundColor(SkColor resize_background_color) {
+  holder_->SetBackgroundColorWhenClipped(resize_background_color);
+}
+
 void WebView::SetCrashedOverlayView(View* crashed_overlay_view) {
   if (crashed_overlay_view_.view() == crashed_overlay_view) {
     return;
diff --git a/ui/views/controls/webview/webview.h b/ui/views/controls/webview/webview.h
index 23ddefe2a7..5a7eea658e 100644
--- a/ui/views/controls/webview/webview.h
+++ b/ui/views/controls/webview/webview.h
@@ -93,6 +93,10 @@ class WEBVIEW_EXPORT WebView : public View,
   void EnableSizingFromWebContents(const gfx::Size& min_size,
                                    const gfx::Size& max_size);
 
+  // Set the background color to use while resizing with a clip. This is white
+  // by default.
+  void SetResizeBackgroundColor(SkColor resize_background_color);
+
   // If provided, this View will be shown in place of the web contents
   // when the web contents is in a crashed state. This is cleared automatically
   // if the web contents is changed. The passed-in overlay view must be owned by
